common/src/main/java/io/netty/util/concurrent/Promise.java

```diff
@@ -19,63 +19,61 @@
  * Special {@link Future} which is writable.
  */
 public interface Promise<V> extends Future<V> {
-
     /**
-     * Marks this future as a success and notifies all
-     * listeners.
+     * Create a new unfulfilled promise.
      *
+     * @param executor The {@link EventExecutor} the promise should be bound to. Completion callbacks will run on this
+     *                 event executor. It is assumed this executor will protect against {@link StackOverflowError}
+     *                 exceptions. The executor may be used to avoid {@link StackOverflowError} by executing a {@link
+     *                 Runnable} if the stack depth exceeds a threshold.
+     * @param <T>      The result type of the promise.
+     * @return The new promise.
+     */
+    static <T> Promise<T> newPromise(EventExecutor executor) {
```

Comment:
Dunno if we actually want to do this. It doesn't seem very compelling, since there'd be few callers to it. (Most callsites will get promises from the channel.)